### PR TITLE
CB-11868 Create DistroX internal API for COD usecase

### PR DESCRIPTION
**Pull request by foldik**
…ontrollers.

See detailed description in the commit message.

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/BlueprintV4Controller.java [hunk 1]
```diff
@@ -68,6 +68,13 @@ public BlueprintV4Response getByName(Long workspaceId, @ResourceName @NotNull St
         return converterUtil.convert(blueprint, BlueprintV4Response.class);
     }
 
+    @Override
+    @InternalOnly
+    public BlueprintV4Response getByNameInternal(Long workspaceId, @AccountId String accountId, @NotNull String name) {
+        Blueprint blueprint = blueprintService.getByWorkspace(NameOrCrn.ofName(name), restRequestThreadLocalService.getRequestedWorkspaceId());
+        return converterUtil.convert(blueprint, BlueprintV4Response.class);
+    }
+
```
Comment (horadla23) [inline core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/BlueprintV4Controller.java:77]: this is the same logic as in `getByName`, can you reuse that?

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/RecipesV4Controller.java [hunk 2]
```diff
@@ -72,6 +72,13 @@ public RecipeV4Response getByName(Long workspaceId, @ResourceName String name) {
         return converterUtil.convert(recipe, RecipeV4Response.class);
     }
 
+    @Override
+    @InternalOnly
+    public RecipeV4Response getByNameInternal(Long workspaceId, @AccountId String accountId, @NotNull String name) {
+        Recipe recipe = recipeService.get(NameOrCrn.ofName(name), restRequestThreadLocalService.getRequestedWorkspaceId());
+        return converterUtil.convert(recipe, RecipeV4Response.class);
+    }
+
```
Comment (horadla23) [inline core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/RecipesV4Controller.java:81]: same here

**Review by horadla23 — CHANGES_REQUESTED** [review]

File: core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/BlueprintV4Controller.java (hunk 1 above)
Comment (foldik) [reply core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/BlueprintV4Controller.java:77]: Ohh, sh.. I was blind. Sure.

Comment (horadla23) [reply core/src/main/java/com/sequenceiq/cloudbreak/controller/v4/BlueprintV4Controller.java:77]: it's not a big deal:) thx!

**Review by horadla23 — APPROVED** [review]

**Pull request merged**